com.woltlab.wcf/templates/benchmark.tpl

```diff
@@ -1,40 +1,49 @@
 	
 	{if ENABLE_DEBUG_MODE}
-		<script data-relocate="true">
-			$(function() {
-				$('#benchmarkDetails > li > span').click(function() {
-					$(this).parent().children('pre').toggle();
-				});
-			});
-		</script>
-		<ul id="benchmarkDetails" style="display: none;">
-			<li id="benchmarkItemPhp" style="margin-bottom: 8px;"{if $__wcf->getBenchmark()->getOffsetToRequestTime() >= 0.5} class="hot"{/if}>
-				<span>Request Execution Time: <kbd>{$__wcf->getBenchmark()->getRequestExecutionTime()}s</kbd></span><br>
-				<small>Benchmark started after: <kbd>{$__wcf->getBenchmark()->getOffsetToRequestTime()}s</kbd></small>
-				<hr>
-			</li>
-			{foreach from=$__wcf->getBenchmark()->getItems() key=benchmarkIndex item=item}
-				<li id="benchmarkItem{@$benchmarkIndex}" style="margin-bottom: 8px;"{if $item.use >= 0.01} class="hot"{/if}>
-					<span>{if $item.type == 1}(SQL Query) {/if}{$item.text}</span><br>
-					<small style="font-size: .85em">Execution time: <kbd>{@$item.use}s</kbd></small>
-				
-				<pre style="display: none">{foreach from=$item.trace key=traceNo item=traceItem}#{#$traceNo} {if !$traceItem.class|empty}{$traceItem.class}{$traceItem.type}{else}{if !$traceItem.file|empty}{$traceItem.file}: {/if}{/if}{$traceItem.function}() {if !$traceItem.line|empty}(line {#$traceItem.line}){/if} 
-{/foreach}</pre>
-				
-				</li>
-			{/foreach}
-		</ul>
+		<div class="benchmarkDetails" hidden>
+			<dl{if $__wcf->getBenchmark()->getOffsetToRequestTime() >= 0.5} class="hot"{/if}>
+				<dt>Request Execution time</dt>
+				<dd>{#$__wcf->getBenchmark()->getRequestExecutionTime()}s</dd>
+				<dt>Benchmark started after</dt>
+				<dd>{#$__wcf->getBenchmark()->getOffsetToRequestTime()}s</dd>
+			</dl>
+			<hr>
+			<ol class="nativeList">
+				{foreach from=$__wcf->getBenchmark()->getItems() key=benchmarkIndex item=item}
+					<li {if $item.use >= 0.01} class="hot"{/if}>
+						<details>
+							<summary>
+								<span>{if $item.type == 1}(SQL Query) {/if}{$item.text}</span><br>
+								<small style="font-size: .85em">Execution time: <kbd>{@$item.use}s</kbd></small>
+							</summary>
+					
+							<ol class="nativeList" start="0">
+								{foreach from=$item.trace item=traceItem}
+									<li>
+										{if !$traceItem.class|empty}{$traceItem.class}{$traceItem.type}{else}{if !$traceItem.file|empty}{$traceItem.file}: {/if}{/if}{$traceItem.function}() {if !$traceItem.line|empty}(line {#$traceItem.line}){/if}
+									</li>
+								{/foreach}
+							</ol>
+					</li>
+				{/foreach}
+			</ol>
+		</div>
 		
 		<script data-relocate="true">
-			$(function() {
-				$('#benchmark').click(function() {
-					$('#benchmarkDetails').wcfDialog({
-						title: 'Log'
-					});
+			require(['WoltLabSuite/Core/Component/Dialog'], ({ dialogFactory }) => {
+				const details = document.querySelector('.benchmark .benchmarkDetails');
+				const dialog = dialogFactory().fromElement(details).withoutControls();
+
+				details.hidden = false;
```

Comment:
I wonder if `dialogFactory.fromElement()` should include some smartness to do this?

Reply:
Yes, setting `<element>.hidden = false;` makes sense.

Reply:
#5315